stdlib/src/collections/list.mojo

```diff
@@ -418,6 +419,25 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
             earlier_idx += 1
             later_idx -= 1
 
+    # TODO: Modify this to be regular method when issue 1876 is resolved
+    @staticmethod
+    fn index[C: ComparableCollectionElement](self: List[C], owned value: C, start: Int = 0, end: Optional[Int] = None) raises -> Int:
+        """Returns the index of the first occurrence of a value in a list; raises Error if not found."""
+        var normalized_start = (self.size + start) if start < 0 else start
+        var normalized_end = self.size if end is None else (self.size + end if end < 0 else (self.size if end > self.size else end))
+
+        if not self.size: raise "Cannot find index of a value in an empty list."
+        if normalized_start >= self.size: raise "Given 'start' parameter (" + String(normalized_start) + ") is out of range. List only has " + String(self.size) + " elements."
+
+        var ret_val: Int
+        for i in range(normalized_start, normalized_end):
+            # Note: Implementing __contains__ with O(n) time complexity in future, indicating it relies on linear search,
+            # could degrade the performance of the List.index method if it were to replace the current implementation,
+            # as it would essentially perform the same loop twice.
```

Comment:
Um yeah. This suggests that if want to first check `value` availability in the list using `in`, this makes the algorithm iterate over the list two times if the value is available. One time for checking value availability and the second time for finding its index. In the first version which was not clean and simple at all, I thought these comments may lead to preventing future "refactors" that can corrupt the performance. Since it's much more understandable now, I also assume that's unnecessary, so I will remove them.